packages/eventual-send/src/E.js

```diff
@@ -171,6 +180,13 @@ const makeEGetProxyHandler = (x, HandledPromise) =>
  * @param {HandledPromiseConstructor} HandledPromise
  */
 const makeE = HandledPromise => {
+  // Note the use of `freeze` rather than `harden` below. This is because
+  // `harden` now implies no-trapping, and we depend on proxies with these
+  // almost-empty targets to remain trapping for traps `get`, `apply`, and `set`
+  // which can still be interesting even when the target is frozen.
+  // `get` and `has`, if not naming an own property, are still general traps,
+  // which we rely on. `apply`, surprisingly perhaps, is free to ignore the
+  // target's call behavior and just do its own thing instead.
```

Comment:
```suggestion
  // Note the use of `freeze` of the Proxy targets below. This is because
  // `harden` now implies non-trapping, and we depend on proxies with these
  // almost-empty targets to remain trapping for traps `get`, `apply`, and `set`
  // which can still be interesting even when the target is frozen.
  // `get` and `has`, if not naming an own property, are still general traps,
  // which we rely on. `apply`, surprisingly perhaps, is free to ignore the
  // target's call behavior and just do its own thing instead.
```

Reply:
Explanation moved and reworked at https://github.com/endojs/endo/pull/2679/files#diff-96bb1b0ba7133e217b7a5114fe5fd34af40a92b43c7b5911a4886ce6554f0dd6 preparing-for-stabilize.md

Please have a look and review there. Resolving here.